clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -221,7 +265,7 @@ public boolean isReady(Node node, long now) {
      * @param node The node
      */
     private boolean canSendRequest(String node) {
-        return connectionStates.isConnected(node) && selector.isChannelReady(node) && inFlightRequests.canSendMore(node);
+        return this.requiredApiVersions == null ? connectionStates.isConnected(node) : connectionStates.isReady(node) && selector.isChannelReady(node) && inFlightRequests.canSendMore(node);
```

Comment:
I do not have a strong bias on this, will be happy to make the change if you really think it is a must.